draft-ietf-tls-tls13.md

```diff
@@ -5142,13 +5242,50 @@ to avoid leaking information about the identities due to length.
 The client's proposed PSK identities are not encrypted, nor is the
 one that the server selects.
 
+### Key Derivation and HKDF
+
+Key derivation in TLS 1.3 uses the HKDF function defined in [RFC 5869] and
+its two components, HKDF-Extract and HKDF-Expand. The full rationale for the HKDF
+construction can be found in [Kraw10] and the rationale for the way it is used 
+in TLS 1.3 in [KW16].  Throughout this document, each
+application of HKDF-Extract is followed by one or more invocations of
+HKDF-Expand. This ordering should always be followed (including in future
+revisions of this document), in particular, one SHOULD NOT use an output of
+HKDF-Extract as an input to another application of HKDF-Extract without an
+HKDF-Expand in between, Consecutive applications of HKDF-Expand are allowed as
+long as these are differentiated via the key and/or the labels.
+
+Note that HKDF-Expand implements a pseudorandom function (PRF) with both inputs and
+outputs of variable length. In some of the uses of HKDF in this document
+(e.g., for generating exporters and the resumption_master_secret), it is necessary
+that the application of HKDF-Expand be collision-resistant, namely, it should
+be infeasible to find two different inputs to HKDF-Expand that output the same
+value. This requires the underlying hash function to be collision resistant
+and the output length from HKDF-Expand to be of size at least 256 bits (or as
+much as needed for the hash function to prevent finding collisions). 
```

Comment:
Some of this paragraph feels like it's duplicating portions of #security-handshake (the PSK binder bits); should some content get trimmed from there?